### PR TITLE
M2P-271 Code refactoring for 3rd-party module Miravist Rewards

### PR DESCRIPTION
**Pull request by johnmccombs1**
# Description
Code refactoring for 3rd-party module Miravist Rewards

Fixes: https://boltpay.atlassian.net/browse/M2P-271

#changelog M2P-271 Code refactoring for 3rd-party module Miravist Rewards

# Type of change

- [ ] Bug fix (change which fixes an issue)
- [ ] New feature (change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?
Please validate that you have tested your change in at least one of the following areas:

- [ ] Successfully tested locally (or docker image)
- [ ] Successfully tested on a staging or sandbox server
- [ ] Successfully tested on a merchant's staging server

# For PR Reviewer 
- [ ] Reviewed unit tests to make sure we are using real components rather than mocks as much as possible?
- [ ] For any major change (observer, new Bolt feature, core Magento interaction) we must add a feature switch, did you verify this?

# Checklist:

- [ ] My code follows the style guidelines of this project.
- [ ] I have performed a self-review of my own code.
- [ ] I have commented my code, particularly in hard-to-understand areas.
- [ ] New and existing unit tests pass locally with my changes.
- [ ] I have created or modified unit tests to sufficiently cover my changes.
- [ ] I have added my Jira ticket link and provided a changelog message.

### CODE REVIEW

File: ThirdPartyModules/Mirasvit/Rewards.php [hunk 1]
```diff
@@ -0,0 +1,185 @@
...
+        $this->bugsnagHelper = $bugsnagHelper;
+        $this->discountHelper = $discountHelper;
+        $this->scopeConfigInterface = $scopeConfigInterface;
+    }
+
+    /**
+     * @param $immutableQuote
+     * @param $mirasvitRewardsPurchaseHelper
+     */
+    public function applyExternalDiscountData($immutableQuote, $mirasvitRewardsPurchaseHelper)
```
Comment (dumega) [inline ThirdPartyModules/Mirasvit/Rewards.php:69]: It should be
```
public function applyExternalDiscountData($mirasvitRewardsPurchaseHelper, $immutableQuote)
```

Or there is a fatal error

```
Fatal Error: 'Uncaught Error: Call to undefined method Mirasvit\\Rewards\\Helper\\Purchase::getBoltParentQuoteId()
```

**Review by dumega — CHANGES_REQUESTED** [review]

File: ThirdPartyModules/Mirasvit/Rewards.php (hunk 1 above)
Comment (johnmccombs1) [reply ThirdPartyModules/Mirasvit/Rewards.php]: @dumega Good catch. I've updated the code. Could you please re-check?

**Review by danac-gs — DISMISSED** [review]

**Review by dumega — DISMISSED** [review]

**Comment by vitaliyreznikov** [comment]
@johnmccombs1 Could you please fix conflicts?

**Comment by johnmccombs1** [comment]
@vitaliyreznikov I've fixed the conflicts

**Comment by danac-gs** [comment]
Can you add unit tests?

**Comment by johnmccombs1** [comment]
@danac-gs   was able to add a unit test for the class ThirdPartyModules/Mirasvit/Rewards.php as in this PR #928. However, Vitaly mentioned that it's not useful

@danac-gs  @vitaliyreznikov @matt-thomason Thought on this?

**Review by dumega — COMMENTED** [review]
@johnmccombs1 Please review https://github.com/BoltApp/bolt-magento2/pull/961 to see if it is ready to merge into yours.

**Comment by johnmccombs1** [comment]
@dumega I've reviewed and merged it

**Review by danac-gs — DISMISSED** [review]

**Comment by vitaliyreznikov** [comment]
@johnmccombs1 Could you please fix the conflict?

**Comment by johnmccombs1** [comment]
@vitaliyreznikov I've fixed the conflict.

**Review by danac-gs — APPROVED** [review]

**Pull request merged**